File: 3471. Find the Largest Almost Missing Integer/Solution.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        n = len(nums)
        
        #count how many subarrays of size k each number appears in
        freq = defaultdict(int)
        
        #for each subarray of size k
        for i in range(n - k + 1):
            #use a set to count each number only once per subarray
            seen = set()
            for j in range(i, i + k):
                if nums[j] not in seen:
                    seen.add(nums[j])
                    freq[nums[j]] += 1
        
        #find the largest number that appears in exactly one subarray
        result = -1
        for num, count in freq.items():
            if count == 1:
                result = max(result, num)
        
        return result

File: 3471. Find the Largest Almost Missing Integer/Solution.cpp
class Solution {
public:
    int largestInteger(std::vector<int>& nums, int k) {
        int n = nums.size();
        std::unordered_map<int, int> freq;
        
        //count how many subarrays of size k each number appears in
        for (int i = 0; i <= n - k; i++) {
            std::unordered_set<int> seen;
            for (int j = i; j < i + k; j++) {
                if (seen.find(nums[j]) == seen.end()) {
                    seen.insert(nums[j]);
                    freq[nums[j]]++;
                }
            }
        }
        
        //find the largest number that appears in exactly one subarray
        int result = -1;
        for (auto& [num, count] : freq) {
            if (count == 1) {
                result = std::max(result, num);
            }
        }
        
        return result;
    }
};
